gammapy/estimators/points/lightcurve.py

```diff
@@ -151,6 +162,24 @@ def run(self, datasets):
 
             valid_intervals.append([t_min, t_max])
 
+            if self.stack_over_time_interval:
+                name = f"timebin_{t_min.mjd:.0f}_{t_max.mjd:.0f}"
+                dataset_reduced = datasets_to_fit.stack_reduce(name=name)
+                models = datasets.models.copy()
+                for dataset in datasets:
+                    models.reassign(dataset.name, dataset_reduced.name)
+                dataset_reduced.models = models
+                datasets_to_fit = Datasets([dataset_reduced])
+
+                if self.n_jobs == 1:
+                    fp = self.estimate_time_bin_flux(
+                        datasets_to_fit, datasets_to_fit.names
+                    )
+                    rows.append(fp)
+                else:
+                    parallel_datasets.append(datasets_to_fit)
+                continue
```

Comment:
Could you avoid redefining the if/else statement and just redefine `dataset_names` ?